CMakePresets.json

```diff
@@ -0,0 +1,60 @@
+{
+    "version": 5,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 24,
+        "patch": 0
+    },
+    "configurePresets": [
+        {
+            "name": "base-dev",
+            "binaryDir": "${sourceDir}/build/${presetName}",
+            "hidden": true
+        },
+        {
+            "name": "msvc-17",
+            "inherits": "base-dev",
+            "displayName": "Visual Studio 2022",
+            "generator": "Visual Studio 17 2022",
```

Comment:
Is anything here specific to this version of Visual Studio besides the generator name? Any reason not to use the Ninja generator?

Reply:
For ninja I can add this:
```json
        {
            "name": "ninja-debug",
            "inherits": "base-dev-ninja",
            "displayName": "Debug Ninja",
            "description": "Build in Debug with Ninja generator",
            "cacheVariables": {
                "CMAKE_BUILD_TYPE": "Debug"
            }
        },
        {
            "name": "ninja-relwithdebinfo",
            "inherits": "base-dev-ninja",
            "displayName": "RelWithDebInfo Ninja",
            "description": "Build in RelWithDebInfo with Ninja generator",
            "cacheVariables": {
                "CMAKE_BUILD_TYPE": "RelWithDebInfo"
            }
        },
        {
            "name": "ninja-release",
            "inherits": "base-dev-ninja",
            "displayName": "Release Ninja",
            "description": "Build in Release with Ninja generator",
            "generator": "Ninja",
            "cacheVariables": {
                "CMAKE_BUILD_TYPE": "Release"
            }
        },
        {
            "name": "ninja-multi",
            "inherits": "base-dev",
            "displayName": "Ninja Multi-Config",
            "description": "Default build using Ninja Multi-Config generator",
            "generator": "Ninja Multi-Config"
        },
```
but the goal of this pr was just to add a preset for a visual studio solution

Reply:
Note that before we did try adding all the combinations of CMakePresets but it kinda exploded into many possibilities #325, with varying options of Qt 5 vs 6, Ninja, vcpkg etc

Reply:
I see that now.
I think it's always nice to have a default preset that "just work" for testing purpose and easy onboarding. Then people can hack their own way.
So having a default with Qt6 let's say is ok.

So from your PR I would suggest creating a base config with:
"CMAKE_CXX_COMPILER_LAUNCHER": "sccache"
"RUSTC_WRAPPER": "sccache"
"USE_QT5": false
"VCPKG": true

I feel it is always better to have documentation like that, than outdated Readme information.
What you be even better is for the presets to be exactly what is being used in CI. So one preset for MSVC, one for linux with ninja, one for mac (with ninja too?), so the same as linux maybe.

Reply:
That's why going the CI way is a good option where there is no debate. If people want something else, either they create custom presets, or use cmake manually. CMakePresets should just provide a starting point, and show all the variable people usually tweak?